src/insert.rs

```diff
@@ -81,7 +81,11 @@ impl SupabaseClient {
     /// # Returns
     /// This method returns a `Result<String, String>`. On success, it returns `Ok(String)` with the new row's ID,
     /// and on failure, it returns `Err(String)` with an error message.
-    pub async fn insert(&self, table_name: &str, mut body: Value) -> Result<String, String> {
+    pub async fn insert_with_generated_id(
```

Comment:
Honestly this method should be removed, or at least deprecated.
The value of the primary key should be either explicitly provided by the user application or generated by the database.